af/fastpath/fastpath/core.py

```diff
@@ -1679,35 +1696,55 @@ def core():
         clean_caches()
 
 
-def setup_fingerprints():
-    """Setup fingerprints lookup dictionary
-    "ZZ" applies a fingerprint globally
-    """
-    # pre-process fingerprints to speed up lookup
+def extract_expected_countries(ec):
+    ecs = set(cc.upper() for cc in ec.strip().split(","))
+    ecs.discard("")
+    if not ecs:
+        ecs = ["ZZ"]
+    return sorted(ecs)
+
+
+@metrics.timer("prepare_fingerprints")
+def prepare_fingerprints(dns_fp, http_fp):
+    for fp in dns_fp + http_fp:
+        exp = set(fp["expected_countries"].strip().split(","))
+        exp.discard("")
+        # Discard ZZ as we don't handle it currently
+        exp.discard("ZZ")
```

Comment:
I think that since in 
```
def is_confirmed(probe_cc: str, fp: Fingerprint) -> bool:
    """Checks if a matching fingerprint represents confiremed blocking"""
    if probe_cc != "ZZ" and probe_cc in fp["expected_countries"]:
```

You can probably drop these two lines where you discard `ZZ`. Once the PR cleaning up the list is, probably the same thing can be done for the previous discard.

Reply:
I'm doing the discard twice on purpose because it has different meaning (cleaning up the global variable VS ignoring msmts in ZZ), given that we are planning to change the matching criteria in future.